modules/SofaSparseSolver/src/SofaSparseSolver/CSR_to_adj.h

```diff
@@ -0,0 +1,15 @@
+#ifndef SOFA_CSR_TO_ADJ_H
+#define SOFA_CSR_TO_ADJ_H
+
+#include <SofaSparseSolver/config.h>
+#include <SofaBaseLinearSolver/MatrixLinearSolver.h>
+
+extern "C" {
+#include <metis.h>
+}
+
+namespace sofa::component::linearsolver
+{
+void CSR_to_adj(int n,int * M_colptr,int * M_rowind,type::vector<int> adj,type::vector<int> xadj); // compute the adjency matrix in CSR format from the matrix given in CSR format
```

Comment:
Excellent! you have isolated this piece of code, now use it in the solvers.

```suggestion
void CSR_to_adj(int n, int * M_colptr, int * M_rowind, type::vector<int>& adj, type::vector<int>& xadj); // compute the adjency matrix in CSR format from the matrix given in CSR format
```